src/plugin-main.c

```diff
@@ -1,342 +1,333 @@
+
+		if (context->effect) {
+			gs_effect_destroy(context->effect);
+		}
+
+		bfree(context);
+
+		obs_leave_graphics();
+	}
+}
+
+static bool filter_reload_effect(void *data)
+{
+	struct filter_9slice *context = data;
+
+	obs_enter_graphics();
+
+	// cleanup any existing effect.
+	if (context->effect) {
+		gs_effect_destroy(context->effect);
+		context->effect = NULL;
+		obs_log(LOG_INFO, "Destroyed existing effect.");
+	}
+
+	char *effect_file = obs_module_file("sliced.hlsl");
+	context->effect = gs_effect_create_from_file(effect_file, NULL);
+	bfree(effect_file);
+
+	struct filter_9slice_params_t *params = &context->params;
+	params->border = NULL;
+	params->output_size = NULL;
+	params->source_size = NULL;
+	params->use_linear_filtering = NULL;
+	params->show_uvs = NULL;
+
+	if (!context->effect) {
+		obs_log(LOG_ERROR, "Failed to create effect.");
+	} else {
+		const gs_effect_t *effect = context->effect;
+
+		params->show_uvs =
+			gs_effect_get_param_by_name(effect, "show_uvs");
+		params->use_linear_filtering = gs_effect_get_param_by_name(
+			effect, "use_linear_filtering");
+		params->border = gs_effect_get_param_by_name(effect, "border");
+		params->output_size =
+			gs_effect_get_param_by_name(effect, "output_size");
+		params->source_size =
+			gs_effect_get_param_by_name(effect, "source_size");
+
+		if (params->use_linear_filtering == NULL) {
+			obs_log(LOG_WARNING,
+				"Failed to get use_linear_filtering param.");
+		}
+		if (params->show_uvs == NULL) {
+			obs_log(LOG_WARNING, "Failed to get show_uvs param.");
+		}
+		if (params->border == NULL) {
+			obs_log(LOG_WARNING, "Failed to get border param.");
+		}
+		if (params->output_size == NULL) {
+			obs_log(LOG_WARNING,
+				"Failed to get output_size param.");
+		}
+		if (params->source_size == NULL) {
+			obs_log(LOG_WARNING,
+				"Failed to get source_size param.");
+		}
+	}
+
+	obs_leave_graphics();
+
+	return context->effect != NULL;
+}
+
+struct cb_data {
+	obs_source_t *parent;
+	obs_sceneitem_t *found;
+};
+typedef struct cb_data cb_data_t;
+
+static bool enum_scenes_callback(void *data_ptr, obs_source_t *scene_source)
+{
+	cb_data_t *data = data_ptr;
+
+	obs_scene_t *scene = obs_group_or_scene_from_source(scene_source);
+	if (!scene)
+		return true;
+
+	obs_sceneitem_t *found = obs_scene_find_source_recursive(
+		scene, obs_source_get_name(data->parent));
+	if (!found)
+		return true;
+
+	// found the scene-item that contain our source!
+	data->found = found;
+	return false;
+}
+
+static void filter_render(void *data, gs_effect_t *effect)
+{
+	UNUSED_PARAMETER(effect);
+
+	struct filter_9slice *context = data;
+	if (!context) {
+		return;
+	}
+
+	obs_source_t *parent = obs_filter_get_parent(context->source);
+
+	struct cb_data target = {
+		.parent = parent,
+		.found = NULL,
+	};
+	obs_enum_scenes(enum_scenes_callback, &target);
+
+	struct vec2 scale = {.x = 1.0f, .y = 1.0f};
+	if (target.found) {
+		obs_sceneitem_get_scale(target.found, &scale);
+	}
```

Comment:
Point is this lines. we can obtain scale factor from scene item that search with parent source name.